core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceManagerResourceMonitorRecordComponent.scala

```diff
@@ -148,7 +153,19 @@ final case class WorkspaceManagerResourceMonitorRecord(
   userEmail: Option[String],
   createdTime: Timestamp,
   args: Option[Map[String, String]] = None
-)
+) {
+
+  def retryOrTimeout(onTimeout: () => Future[Any] = () => Future.successful())(implicit
+    executionContext: ExecutionContext
+  ): Future[JobStatus] = {
+    val expireTime = Instant.ofEpochMilli(createdTime.getTime).plus(1, ChronoUnit.DAYS)
```

Comment:
Do we want any logging in this method?

Reply:
I think it makes sense to leave it to the call site to do any logging, just like we're leaving it to the call site to record the error if appropriate.